fetch_worker.go

```diff
@@ -71,21 +90,45 @@ func (w *fetchWorker) run(ctx context.Context) (err error) {
 
 	defer func() { err = errors.Join(err, tx.Commit()) }()
 
-	start, end := w.start, w.end
-	for {
-		rows, err := w.selectRowsChunk(ctx, tx, start, end)
+	sdk.Logger(ctx).Info().
+		Uint64("start", w.start).
+		Uint64("end", w.end).
+		Uint64("fetchSize", w.config.fetchSize).
+		Msg("fetching rows")
+
+	for chunkStart := w.start; chunkStart <= w.end; chunkStart += w.config.fetchSize {
+		chunkEnd := chunkStart + w.config.fetchSize
+		sdk.Logger(ctx).Info().
+			Uint64("chunk start", chunkStart).
+			// the where clause is exclusive at the end
```

Comment:
It is exclusive at the end to mimic how do we normally do loops (i := 0; i < thing.length; i++). I don't know what's more intuitive though, I wrote that -1 so that logs are less confusing.

Reply:
@hariso I'd like to actually address this in another pr, as when approaching making this simpler I'm seeing other stuff that I want to refactor aswell:

![image](https://github.com/user-attachments/assets/f613b35c-ea90-44f7-84bc-3f7877e36fb9)

(look at the ids - username).

Reply:
Actually I'm going to address this in this pr aswell, as it affects the clarity of other comments.